roofit/roofitcore/src/TestStatistics/RooBinnedL.cxx

```diff
@@ -0,0 +1,163 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+/**
+\file RooBinnedL.cxx
+\class RooBinnedL
+\ingroup Roofitcore
+
+Class RooBinnedL implements a a -log(likelihood) calculation from a dataset
+and a PDF. The NLL is calculated as
+<pre>
+ Sum[data] -log( pdf(x_data) )
+</pre>
+In extended mode, a (Nexpect - Nobserved*log(NExpected) term is added
+**/
+
+#include <TestStatistics/RooBinnedL.h>
+
+#include "RooAbsData.h"
+#include "RooAbsPdf.h"
+#include "RooAbsDataStore.h"
+#include "RooRealSumPdf.h"
+#include "RooRealVar.h"
+
+#include "TMath.h"
+
+namespace RooFit {
+namespace TestStatistics {
+
+RooBinnedL::RooBinnedL(RooAbsPdf* pdf, RooAbsData* data) :
```

Comment:
Doxygen. I'll keep saying it, because it was so much work to get to the state that RF is in now ...